doc/site/docs/02-core-concepts/06-body.md

```diff
@@ -0,0 +1,507 @@
+---
+sidebar_position: 6
+sidebar_label: Body
+---
+
+# `Body` Class
+
+The Body system in Relic provides a powerful and flexible way to handle HTTP request and response bodies. It supports various content types, automatic MIME type detection, encoding handling, and efficient streaming for large payloads.
+
+The `Body` class is the core component that represents the content of HTTP messages, designed to handle both small and large payloads efficiently through streaming, making it suitable for everything from simple text responses to large file uploads.
+
+## Key Features
+
+The Body class offers several important capabilities that make it well-suited for modern web applications:
+
+- **Stream-based architecture**: All body content is handled as `Stream<Uint8List>` for memory efficiency, allowing you to process large files without loading them entirely into memory
+- **One-time read constraint**: Bodies can only be read once to prevent accidental double-consumption and ensure predictable behavior
+- **Automatic type detection**: Intelligent MIME type inference for common formats like JSON, HTML, XML, and various binary file types
+- **Encoding support**: Handles text encoding (UTF-8, Latin-1, etc.) automatically while allowing manual override when needed
+
+## Creating Bodies
+
+### From String Content
+
+The most common way to create a body is from string content. Relic provides intelligent automatic detection of content types based on the string's structure, though you can always override this behavior when needed.
+
+For simple text content, you can create a body directly from a string:
+
+```dart
+final body = Body.fromString('Hello, World!');
+```
+
+When you provide JSON content, Relic automatically detects the MIME type by analyzing the string structure:
+
+```dart
+final jsonBody = Body.fromString('{"message": "Hello"}');
+// Automatically detects application/json MIME type
+```
+
+Similarly, HTML content is automatically recognized and tagged with the appropriate MIME type:
+
+```dart
+final htmlBody = Body.fromString('<!DOCTYPE html><html>...</html>');
+// Automatically detects text/html MIME type
+```
+
+For cases where you need explicit control over the MIME type and encoding, you can specify these parameters directly:
```

Comment:
Also, while we support detection, you should prefer setting it. We waste a few cpu cycles to do the detection.

Reply:
Maybe not here, but perhaps mention it in an insert somewhere. It is fast, fairly precise wrt. binary types, does rudimentary detection of text/* (ie. json, xml, and html). Still, if you know the mime-type, prefer to set it explicitly.

Reply:
<img width="873" height="282" alt="Screenshot 2025-10-20 at 11 04 50 AM" src="https://github.com/user-attachments/assets/2763a097-f5e5-4c11-924d-eb0152ff67ba" />
Added this